src/consensus/ledger_enclave_types.h

```diff
@@ -30,6 +30,7 @@ namespace consensus
     /// Modify the local ledger. Enclave -> Host
     DEFINE_RINGBUFFER_MSG_TYPE(ledger_append),
     DEFINE_RINGBUFFER_MSG_TYPE(ledger_truncate),
+    DEFINE_RINGBUFFER_MSG_TYPE(ledger_compact),
```

Comment:
Minor: I wonder if this should be `ledger_committed`? The enclave's responsibility ends at telling the host that a specific committ-_able_ entry has now been committ-_ed_, and it doesn't care what the host does with this information. We currently use "compact" as the general term here but it means different things for different systems. In particular, every where within the enclave it means "forget about data from before this index", which is very much not what the host should be doing.

Reply:
Agreed, `ledger_committed`, or maybe `ledger_freeze` or `ledger_make_read_only` if we want to only express the file consequences and leaved out the consensus implications (technically what matter is, this is now frozen, we won't go back on it. It's because it's committed, but that's the reason, not the action we're telling the host to take).

Reply:
I think `ledger_committed` is the best option. This message does not necessarily mean that the chunk that contains this index will be frozen/read-only from then on as that chunk might not yet be complete.

Reply:
But the message applies to an index, not a chunk? I does mean that all entries in the chunk(s) up to that point are frozen/read-only. Anyway, committed is good too.